pkg/knowledge_base/edge_kb.go

```diff
@@ -402,3 +403,41 @@ func (kb EdgeKB) findPathsInGraph(source, dest core.Resource, stack []graph.Edge
 	delete(visited, source)
 	return result
 }
+
+func (kb EdgeKB) GetTrueUpstream(source core.Resource, dag *core.ResourceGraph) []core.Resource {
```

Comment:
Can you add docs for `GetTrueUpstream` and `GetTrueDownstream` to make their purpose clear?

Reply:
yeah going to add docs to the new pr